hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateHandler.java

```diff
@@ -380,70 +362,145 @@ private void updateExpiryFields(
-            builder.pauseKey(op.pauseKey());
+        if (op.hasAutoRenewAccount()) {
+            context.requireKeyOrThrow(op.autoRenewAccountOrThrow(), INVALID_AUTORENEW_ACCOUNT);
         }
-        if (op.hasMetadataKey()) {
-            validateTrue(originalToken.hasMetadataKey(), TOKEN_HAS_NO_METADATA_KEY);
-            builder.metadataKey(op.metadataKey());
+
+        // if we remove the admin key or any of the low priority keys, we should allow
+        // only the admin key to sign.
+        if (containsKeyRemoval(op)) {
+            validateTruePreCheck(originalToken.hasAdminKey(), INVALID_SIGNATURE);
+            context.requireKey(originalToken.adminKey());
+            return;
         }
 
-        if (isMetadataOnlyUpdateOp(op)) {
-            validateTrue(originalToken.hasAdminKey() || originalToken.hasMetadataKey(), TOKEN_IS_IMMUTABLE);
-        } else if (!isExpiryOnlyUpdateOp(op)) {
-            validateTrue(originalToken.hasAdminKey(), TOKEN_IS_IMMUTABLE);
+        List<Key> lowPriorityKeys;
+        if (originalToken.hasAdminKey()) {
+            // if we update any of the low priority keys, we should allow either admin key
+            // or the respective low priority key to sign.
+            if (noOtherFieldThanLowPriorityKeyOrMetadataWillBeUpdated(op)) {
+                lowPriorityKeys = getAllRequiredLowPrioritySigners(originalToken, op);
+                Key requiredKey;
+                if (lowPriorityKeys.isEmpty()) {
+                    requiredKey = originalToken.adminKey();
+                } else {
+                    final Key lowPriorityKeyList = Key.newBuilder()
+                            .keyList(KeyList.newBuilder().keys(lowPriorityKeys).build())
+                            .build();
+                    final List<Key> keysRequired = List.of(lowPriorityKeyList, originalToken.adminKey());
+                    // with this we will require either admin key signature or all low priority keys signatures
+                    // (we can update several low priority keys with one TokenUpdate, so we will require all of these
+                    // keys
+                    // to sign)
+                    requiredKey = Key.newBuilder()
+                            .thresholdKey(ThresholdKey.newBuilder()
+                                    .keys(KeyList.newBuilder()
+                                            .keys(keysRequired)
+                                            .build())
+                                    .threshold(1)
+                                    .build())
+                            .build();
+                }
+
+                context.requireKey(requiredKey);
+            } else if (!isExpiryOnlyUpdateOp(op)) {
+                // if we update any other field, we need the current admin key to sign
+                // note: for expiry only op admin key is not required
+                context.requireKey(originalToken.adminKey());
+            }
```

Comment:
What happens if Token doesn't have adminKey and its not `expiryOnlyUpdateOp`. Do we still throw somewhere ?

Reply:
@Neeharika-Sompalli 
I believe updating **only** expiry does not need any signatures, so it should succeed or my assumptions is not quite true?

Reply:
No I mean, if it is **not** expiry only op and there is **no** `adminKey`

Reply:
@Neeharika-Sompalli 
In this case:
- if the token has no low priority key set it will result in `TOKEN_IS_IMMUTABLE`
- otherwise check required signatures